### ISSUE TITLE
uistate hash, similar to history.state, is easy to clear

### ISSUE THREAD

**Issue by tbondwilkinson** [slightlyoff/history_api#7]
The current API proposal for uistate hash portion serializes as a string, and the API makes it easy to accidentally overwrite an existing value.

A better API would provide scoping so that other pieces of the page could update the hash without knowing about existing uistate.

**Comment by dvoytenko** [comment]
Agreed. In this case it'd be a requirement since, unlike `:~:text=`, it's more likely to have `:~:uistate=` and fragment at the same time.
